### PR TITLE
Added Jest support

### LINKED ISSUE #182: Add Jest to visualizer
**Is your feature request or enhancement related to a problem? What specifically could be better? Please describe.**
We want to have a unit testing framework for the visualizer.

**Describe the solution you'd like**
I would like to use Jest (https://jestjs.io/). To complete this issue, add the library to the visualizer and create a small test unit test. Future unit test will be added organically as the project matures.

### PR DESCRIPTION
**Pull request by kelvinkoon**
### Description
Added Jest, a testing library for JavaScript. Jest will be used to test to test the components in the Visualizer, such as actions, reducers, and sagas.

### Testing Done
An example test was added. When running 'yarn test', the test passes which shows Jest was successfully added.

### Resolved Issues
resolves #182 

### Review Checklist
*(Please check every item to indicate your code complies with it (by changing `[ ]`->`[x]`). This will hopefully save both you and the reviewer(s) a lot of time!)*
***It is the reviewers responsibility to also make sure every item here has been covered***
- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom` 
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/corner_kick/jest.config.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module.exports = {
```
Comment (qgolsteyn) [inline src/corner_kick/jest.config.js]: Missing top-level comment

File: src/corner_kick/src/__tests__/example.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+test('example jest test', () => {
```
Comment (qgolsteyn) [inline src/corner_kick/src/__tests__/example.ts:8]: Missing top-level comment

File: src/corner_kick/jest.config.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  "testRegex": "(/__tests__/.*|(\\.|/)(test|spec))\\.tsx?$",
+  "moduleFileExtensions": [
+    "ts",
+    "tsx",
+    "js",
+    "jsx",
+    "json",
+    "node"
+  ],
+}
```
Comment (qgolsteyn) [inline src/corner_kick/jest.config.js]: Nit: git complaining about missing newline character at end of file

File: src/corner_kick/jest.config.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+module.exports = {
+  "roots": [
+    "<rootDir>/src"
+  ],
+  "transform": {
+    "^.+\\.tsx?$": "ts-jest"
```
Comment (qgolsteyn) [inline src/corner_kick/jest.config.js]: Maybe some documentation here would be useful. What does this line of code mean?

File: src/corner_kick/src/__tests__/example.ts (hunk 2 above)
Comment (qgolsteyn) [inline src/corner_kick/src/__tests__/example.ts:8]: Missing function comment describing test

**Review by qgolsteyn — CHANGES_REQUESTED** [review]
Overall looks good! Let's try to get all those code comments in.

**Comment by qgolsteyn** [comment]
Other thoughts:
- Would be good to see testing documented in the README
- Can we integrate testing with our CI build? @garethellis0 @MathewMacDougall any thoughts?

**Comment by garethellis0** [comment]
Adding this to CI should be pretty straightforward. It would be good to get others working with the CI system too! @kelvinkoon - I'm happy to sit down and chat about it this week if you like (just won't be around on Saturday), or feel free to just ping me on Slack if you have any questions.

File: src/corner_kick/src/__tests__/example.ts (hunk 2 above)
Comment (kelvinkoon) [reply src/corner_kick/src/__tests__/example.ts:8]: This has been fixed.

File: src/corner_kick/jest.config.js (hunk 3 above)
Comment (kelvinkoon) [reply src/corner_kick/jest.config.js]: All the moduleFileExtensions have been put onto a single line instead.

File: src/corner_kick/jest.config.js (hunk 4 above)
Comment (kelvinkoon) [reply src/corner_kick/jest.config.js]: Added an additional comment.

File: src/corner_kick/src/__tests__/example.ts (hunk 2 above)
Comment (kelvinkoon) [reply src/corner_kick/src/__tests__/example.ts:8]: Added a function about how it is a sanity test to check if Jest is configured.

File: src/corner_kick/config/jest.config.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+ * such as actions and reducers
+ */
+module.exports = {
+    rootDir: '..',
+    roots: ['<rootDir>/src'],
+    // Assigning all tsx files to be interpreted by ts-jest
+    transform: {
+        '^.+\\.tsx?$': 'ts-jest',
+    },
+    // Any files inside __tests__ is a test file
```
Comment (garethellis0) [inline src/corner_kick/config/jest.config.js:14]: Why `__tests__` instead of `tests`? Do the underscores have some significance?

Comment (qgolsteyn) [reply src/corner_kick/config/jest.config.js:14]: Allows folder to be at the top of the parent directory. It's a convention with Jest

**Review by qgolsteyn — DISMISSED** [review]
LGTM :+1:

File: src/corner_kick/config/jest.config.js (hunk 5 above)
Comment (qgolsteyn) [reply src/corner_kick/config/jest.config.js:14]: Also Jest looks specifically for this type of folder name for tests

**Review by MathewMacDougall — DISMISSED** [review]

**Review by MathewMacDougall — APPROVED** [review]

**Review by qgolsteyn — APPROVED** [review]

**Pull request merged**